core/src/main/java/org/jboss/set/mjolnir/archive/GitDeleteRepository.java

```diff
@@ -0,0 +1,43 @@
+package org.jboss.set.mjolnir.archive;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.ListBranchCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.internal.storage.file.FileRepository;
+import org.eclipse.jgit.lib.Ref;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+public class GitDeleteRepository {
+
+    private String PATH;
+    private String gitUserName;
+
+    public GitDeleteRepository(String path, String gitUserName) {
+        this.PATH = path;
+        this.gitUserName = gitUserName;
+
+    }
+
+    public void gitRemoveBranches() throws GitAPIException, IOException {
+        File gitDir = new File(PATH);
+        List<Ref> refs = Git.open(gitDir)
+                .branchList().setListMode(ListBranchCommand.ListMode.REMOTE)
+                .call();
+        FileRepository localRepo = new FileRepository(PATH + "/.git");
+        Git git = new Git(localRepo);
+        refs.forEach(ref -> {
+            try {
+                if (ref.getName().contains(gitUserName)) {
+                    git.branchDelete().setBranchNames(ref.getName()).setForce(true).call();
```

Comment:
Same thing here - extract the branch deletion into a separate method in the `GitArchiveRepository` class. So in effect this class would be merged with `GitArchiveRepository`.

The remaining logic, after extracting code for reading and deleting branches, can be put directly into the batchlet?

Reply:
**Added into GitArchiveRepository**